flycheck-eglot.el

```diff
@@ -85,6 +85,10 @@
   :group 'flycheck-eglot)
 
 
+(defvar flycheck-eglot-mode nil
```

Comment:
Why is this needed?

Reply:
`flycheck-eglot--flymake-diagnostics-wrapper` is defined before minor mode `flycheck-eglot-mode`. There is `reference to free variable 'flycheck-eglot-mode'` warning in it.

Reply:
Got it. Well this and the other changes to appease the. byte compiler should be in a separate commit IMO.

Reply:
Other unrelated changes are reverted.

Reply:
You can have those as a separate commit in the same PR, btw. I just wanted it to be clear those are not part of the fix, but rather something unrelated.